### PR TITLE
ci: make black fail pre-commit rather than succeed and modify file

### PR DESCRIPTION
**Pull request by romainkomorndatadog**
Introduces lint:black_check as a new hatch script, and uses it during pre-commit so that pre-commit fails when black would reformat files, rather than reformat the file, succeed, and leave an uncommitted file behind.

This also tweaks .gitignore to ignore cmake_build which was introduced in 3ad6f126c0e7b97a74f58c2ea66ff368ac273b2c .

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)
- [x] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

### CODE REVIEW

**Comment by romainkomorndatadog** [comment]
Testing (in a comment, so it doesn't pollute the commit message).

With a bad commit:
```
romain.komorn@COMP-CLXFH7L2J3 dd-trace-py % git diff --cached
diff --git a/ddtrace/contrib/pytest/plugin.py b/ddtrace/contrib/pytest/plugin.py
index a71f34dfb..ed8bafd96 100644
--- a/ddtrace/contrib/pytest/plugin.py
+++ b/ddtrace/contrib/pytest/plugin.py
@@ -388,6 +388,7 @@ def pytest_configure(config):
         _CIVisibility.enable(config=ddtrace.config.pytest)


+
 def pytest_sessionstart(session):
     # Just a comment to ignore
     if _CIVisibility.enabled:
```
```
...
───────────────────────────────────────────────────────────────────────────────────────── lint ──────────────────────────────────────────────────────────────────────────────────────────
Finished checking dependencies
cmd [1] | black --check ddtrace/contrib/pytest/plugin.py
would reformat ddtrace/contrib/pytest/plugin.py

Oh no! 💥 💔 💥
1 file would be reformatted.
[Autohook] FINISH 01-black
...
[Autohook] A pre-commit script yielded negative exit code 1
```

With a good commit:
```
romain.komorn@COMP-CLXFH7L2J3 dd-trace-py % git diff --cached
diff --git a/ddtrace/contrib/pytest/plugin.py b/ddtrace/contrib/pytest/plugin.py
index a569c95d8..a71f34dfb 100644
--- a/ddtrace/contrib/pytest/plugin.py
+++ b/ddtrace/contrib/pytest/plugin.py
@@ -389,6 +389,7 @@ def pytest_configure(config):


 def pytest_sessionstart(session):
+    # Just a comment to ignore
     if _CIVisibility.enabled:
         log.debug("CI Visibility enabled - starting test session")
         global _global_skipped_elements
```
```
romain.komorn@COMP-CLXFH7L2J3 dd-trace-py % git commit -am "a good commit"
[Autohook] Looking for pre-commit scripts to run...found 5!
[Autohook] BEGIN 01-black
...
───────────────────────────────────────────────────────────────────────────────────────── lint ──────────────────────────────────────────────────────────────────────────────────────────
Finished checking dependencies
cmd [1] | black --check ddtrace/contrib/pytest/plugin.py
All done! ✨ 🍰 ✨
1 file would be left unchanged.
[Autohook] FINISH 01-black
...
```
exits with 0 and code is committed.

**Comment by romainkomorndatadog** [comment]
Another option here would be to re-commit the file post-black, but I don't like committing things automatically on behalf of the user, so I prefer this.

It's trivial to fix with `hatch run lint:fmt` if the pre-commit hook fails.

**Review by emmettbutler — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -33,6 +33,7 @@ ddtrace/internal/datadog/profiling/libdatadog/
 .Python
 env/
 build/
+cmake_build/
```
Comment (majorgreys) [inline .gitignore:36]: This was recently removed by @sanchda in https://github.com/DataDog/dd-trace-py/commit/4489271007a802605b490f69e980b4344c9dbbc5#diff-bc37d034bad564583790a46f19d807abfe519c5671395fd494d8cce506c42947L36

Do we need it?

Comment (romainkomorndatadog) [reply .gitignore:36]: yeah, not anymore, I'm gonna drop it

Comment (romainkomorndatadog) [inline .gitignore:36]:
```suggestion
```

**Pull request merged**